### PR TITLE
DT-1014 remove elite2api context

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/web/config/ResourceServerConfiguration.java [hunk 1]
```diff
@@ -28,8 +28,8 @@ public void configure(final HttpSecurity http) throws Exception {
                 // Can't have CSRF protection as requires session
                 .and().csrf().disable()
                 .authorizeRequests(auth ->
-                    auth.antMatchers("/webjars/**", "/favicon.ico", "/csrf",
-                        "/health", "/info", "/ping", "/health/ping", "/h2-console/**",
+                    auth.antMatchers("/webjars/**", "/favicon.ico", "/csrf", "/elite2api/**",
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/web/config/ResourceServerConfiguration.java:31]: Needed to allow rewrite to kick in

File: helm_deploy/prison-api/templates/deployment.yaml [hunk 2]
```diff
@@ -48,15 +48,15 @@ spec:
...
+              path: /health/liveness
               port: {{ .Values.image.port }}
             periodSeconds: 30
             initialDelaySeconds: 90
             timeoutSeconds: 20
             failureThreshold: 10
           readinessProbe:
             httpGet:
-              path: /elite2api/ping
+              path: /health/readiness
```
Comment (Mjwillis) [inline helm_deploy/prison-api/templates/deployment.yaml:59]: switch to readiness/liveness

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -1,5 +1,13 @@
 server:
   port: 8080
+  servlet:
+    context-path: /
```
Comment (Mjwillis) [inline src/main/resources/application.yml:4]: removed elite2api context

File: helm_deploy/prison-api/templates/deployment.yaml (hunk 2 above)
Comment (mattops) [reply helm_deploy/prison-api/templates/deployment.yaml:59]: is this a in-built spring boot endpoint?

File: src/main/java/uk/gov/justice/hmpps/prison/web/config/UrlRewriteConfig.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.stereotype.Component;
+import org.tuckey.web.filters.urlrewrite.Conf;
+import org.tuckey.web.filters.urlrewrite.UrlRewriteFilter;
+
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import java.io.IOException;
+
+@Component
+public class UrlRewriteConfig extends UrlRewriteFilter {
```
Comment (matt-crowson-moj) [inline src/main/java/uk/gov/justice/hmpps/prison/web/config/UrlRewriteConfig.java:14]: Just wondering if there is any value in overriding anything else here to see when this is used/called for tracing purposes?

e.g. the doFilter(...) method.

or using aspects or something to add some logging perhaps?

**Pull request merged**